spec/plaindate.html

```diff
@@ -740,13 +747,15 @@ <h1>ToTemporalDate ( _item_ [ , _options_ ] )</h1>
       <emu-alg>
         1. If _options_ is not present, set _options_ to *undefined*.
         1. Assert: Type(_options_) is Object or Undefined.
+        1. If _options_ is not *undefined*, set _options_ to ? SnapshotOwnProperties(? GetOptionsObject(_options_), *null*).
         1. If Type(_item_) is Object, then
           1. If _item_ has an [[InitializedTemporalDate]] internal slot, then
             1. Return _item_.
           1. If _item_ has an [[InitializedTemporalZonedDateTime]] internal slot, then
             1. Perform ? ToTemporalOverflow(_options_).
             1. Let _instant_ be ! CreateTemporalInstant(_item_.[[Nanoseconds]]).
-            1. Let _plainDateTime_ be ? GetPlainDateTimeFor(_item_.[[TimeZone]], _instant_, _item_.[[Calendar]]).
+            1. Let _timeZoneRec_ be ? CreateTimeZoneRecord(_item_.[[TimeZone]], « *"getOffsetNanosecondsFor"* »).
+            1. Let _plainDateTime_ be ? GetPlainDateTimeFor(_timeZoneRec_, _instant_, _item_.[[Calendar]]).
             1. Return ! CreateTemporalDate(_plainDateTime_.[[ISOYear]], _plainDateTime_.[[ISOMonth]], _plainDateTime_.[[ISODay]], _plainDateTime_.[[Calendar]]).
```

Comment:
Is _options_ validated at the right place here? We generally want to process arguments in order per the reasoning behind #2478 and [how-we-work](https://github.com/tc39/how-we-work/pull/119/files#diff-b48c7b3eba9c6b2a43ae502ddc9eabd1a9c49a2aa5592c64ea8ef548dc81352aR18-R26), but it might be conceivable that a future change extends options with something that must get passed through to CreateTimeZoneRecord and/or GetPlainDateTimeFor. Likewise for ToTemporalDateTime/ToTemporalMonthDay/ToTemporalYearMonth/etc.

Related: https://github.com/tc39/proposal-temporal/pull/2574#discussion_r1261802726

Reply:
I'm addressing this in #2645, have a look there and see what you think. (This also revealed a gap in test coverage, since initially when I made the change, only a PlainDate.from test failed.)